### ISSUE TITLE
Stuck on long web summary

### ISSUE THREAD

**Issue by psinger** [huggingface/chat-ui#292]
I am trying the web search functionality and I am trying to pass longer summaries to the LLM endpoint.

However, I am getting stuck if the summary is above a certain length in this part of the code:
https://github.com/huggingface/chat-ui/blob/main/src/routes/conversation/%5Bid%5D/web-search/%2Bserver.ts#L45

It should be able to be replicated if you manually set this line:
https://github.com/huggingface/chat-ui/blob/main/src/routes/conversation/%5Bid%5D/web-search/%2Bserver.ts#L99

to something like:
webSearch.summary = 'a very long text here ...'

Is this a limitation of the POST request? Does it need to be chunked? I am no expert on this, so grateful on any guidance.

cc @nsarrazin

**Comment by nsarrazin** [comment]
Hi! Thanks for the feedback, I'll try to investigate the issue, not sure what might be causing this :thinking:

**Comment by psinger** [comment]
After some digging, the issue might be that long output is written to the `appendUpdate` here:
https://github.com/huggingface/chat-ui/blob/main/src/routes/conversation/%5Bid%5D/web-search/%2Bserver.ts#L100

That said, even after doing that, I see overflow errors on really long text, but it helps some of my issues. Potentially this would need to be chunked in the GET request.

**Comment by nsarrazin** [comment]
Yeah we probably need to chunk it, or simply clip it when passing it to `appendUpdate`  (it will still get passed in full in the backend, the update is only for the UI)

**Comment by psinger** [comment]
Thanks - I also noticed another thing that sometimes the reader is updating json too fast, and it is showing a json parse error here:
https://github.com/huggingface/chat-ui/blob/main/src/routes/conversation/%5Bid%5D/%2Bpage.svelte#L177

It somehow gets concatenated json strings:
`{"messages":[{"type":"update","message":"Generating search query"}]}{"messages":[{"type":"update","message":"Generating search query"},{"type":"update","message":"Searching Google"}]}`

I am unfortunately too unknowledgable here, but seems there is kind of a race condition.
